src/main.py

```diff
@@ -0,0 +1,180 @@
+        self.prompt = Prompt()
+
+    def home_page(self):
+        while True:
+            try:
+                option = self.prompt.home()
+                if option == 1:
+                    username, password = self.prompt.creds()
+                    user = self.user_repo.create_user(
+                        username=username, password=password
+                    )
+                    return user
+                if option == 2:
+                    username, password = self.prompt.creds()
+                    flag, user = authenticate(
+                        username=username,
+                        password=password,
+                    )
+                    if not flag:
+                        self.prompt.login_failed()
+                        self.prompt.end()
+                        continue
+                    return user
+                if option == 3:
+                    self.prompt.end()
+                else:
+                    self.prompt.invalid_input()
+                    logger.info("invalid input, input value: %s", str(option))
+
+            except ValueError as exc:
+                self.prompt.invalid_input()
+                logger.info("Invalid input, Input Value: %s", str(exc))
+
+    def todo_interface(self, user):
+        task_repo = TaskRepository(user=user, session=self.session)
+        while True:
+            try:
+                option = self.prompt.todo_interface()
+                if option == 1:
+                    task_name = self.prompt.task_input()
+                    task = task_repo.create_task(name=task_name)
+                    self.prompt.display_task(task=task)
+
+                if option == 2:
+                    tasks = task_repo.get_all()
+                    self.prompt.display_task(task=tasks)
+
+                if option == 3:
+                    tasks = task_repo.get_all()
+                    self.prompt.display_task(task=tasks)
+                    task_id = self.prompt.input_task_id()
+                    task_name, task_status = self.prompt.update_task_input()
+                    task_repo.update_task(
+                        id=task_id, name=task_name, status=task_status
+                    )
+                    self.prompt.updated_successfully()
+
+                if option == 4:
+                    task_id = self.prompt.input_task_id()
+                    task_repo.delete_task_by_id(id=task_id)
+
+                if option == 5:
+                    self.prompt.end()
+                    break
+            except Exception as exc:
+                print("An error occured", str(exc))
+
+
+def main():
+    try:
+        with Session(
+            engine
+        ) as session:  #  session obj is created using the 'engine'.In SQLALchemy, a session is a way to
+            # interact with db
+            todo = Todo(session=session)  #
+            try:
+                user = todo.home_page()
+
+            except NoUserInDatabase as exc:
```

Comment:
@rajan10 , why are you catching exception here and not in authenticate function ??